roofit/roostats/src/AsymptoticCalculator.cxx

```diff
@@ -1469,14 +1463,14 @@ RooAbsData * AsymptoticCalculator::MakeAsimovData(const ModelConfig & model, con
               cClass != RooGamma::Class() && cClass != RooLognormal::Class() &&
               cClass != RooBifurGauss::Class()  ) {
             TString className =  (cClass) ?  cClass->GetName() : "undefined";
-            oocoutW((TObject*)0,Generation) << "AsymptoticCalculator::MakeAsimovData:constraint term "
+            oocoutW(static_cast<TObject *>(0),Generation) << "AsymptoticCalculator::MakeAsimovData:constraint term "
                                             << cterm->GetName() << " of type " << className
                                             << " is a non-supported type - result might be not correct " << std::endl;
          }
 
          // in case of a Poisson constraint make sure the rounding is not set
          if (cClass == RooPoisson::Class() ) {
-            RooPoisson * pois = dynamic_cast<RooPoisson*>(cterm);
+            RooPoisson * pois = static_cast<RooPoisson*>(cterm);
```

Comment:
Again, one of the cases where the `dynamic_cast` would have only been used in debug mode inside the `assert` below, so you were right to use `static_cast` here!

But that means the `dynamic_cast` has to be done in the assert below:
```C++
assert(dynamic_cast<RooPoisson*>(pois));
```